### PR TITLE
#292: Added option to dump test results to a custom file and enable C…

### PR DESCRIPTION
**Pull request by tapspatel**
…I workflow to summarize ttrt run and perf test results

This PR gives users the ability to dump test results into a json file and allows CI to upload the tests results to summarize for ttrt run and ttrt perf tests.

### CODE REVIEW

**Comment by tapspatel** [comment]
converting to draft while I fix some CI upload issues

**Review by nsmithtt — APPROVED** [review]

File: runtime/tools/python/test/test_check.py [hunk 1]
```diff
@@ -27,151 +28,214 @@
...
     check_instance()
 
     assert (
-        check_results("check_results.json") == 0
-    ), f"one of more tests failed in={test_clean_artifacts.__name__}"
+        check_results(
+            f"ttrt-results/{inspect.currentframe().f_code.co_name}_check.json"
+        )
+        == 0
+    ), f"one of more tests failed in={tinspect.currentframe().f_code.co_name}"
```
Comment (vmilosevic) [inline runtime/tools/python/test/test_check.py:44]: This check seems to repeat in all tests, maybe you can avoid copy paste by making common assert, or maybe 
[pytest_runtest_teardown / pytest_runtest_teardown](https://docs.pytest.org/en/latest/reference/reference.html#test-running-runtest-hooks) hooks can be useful here to keep things [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

File: runtime/tools/python/ttrt/common/util.py [hunk 2]
```diff
@@ -672,3 +672,39 @@ def save_results(self, file_name="results.json"):
...
+            testcase.set("name", item.get("file_path", ""))
+            testcase.set("file_path", item.get("file_path", ""))
+            testcase.set("result", item.get("result", ""))
+            testcase.set("exception", item.get("exception", ""))
+            testcase.set("log_file", item.get("log_file", ""))
+            testcase.set("artifacts", item.get("artifacts", ""))
+            testcase.set("program_index", item.get("program_index", ""))
+
+        tree = ET.ElementTree(testsuites)
+        xml_file_path = "ttrt_report.xml"
```
Comment (vmilosevic) [inline runtime/tools/python/ttrt/common/util.py:709]: Can I see somewhere example of generated report?

File: .github/workflows/build-and-test.yml [hunk 3]
```diff
@@ -242,6 +242,14 @@ jobs:
         source env/activate
         ttrt ${{ matrix.build.name }} ${{ matrix.build.ttrt_flags }} ${{ steps.strings.outputs.build-output-dir }}/test/ttmlir/Silicon --host-only
 
+    - name: Show Test Report
+      uses: mikepenz/action-junit-report@v4
+      if: success() || failure()
+      with:
+        report_paths: ttrt_report.xml
```
Comment (vmilosevic) [inline .github/workflows/build-and-test.yml:249]: Can you also upload ttrt report to artifacts?

**Review by vmilosevic — COMMENTED** [review]
Looks good, but please cleanup copy/paste check

File: .github/workflows/build-and-test.yml (hunk 3 above)
Comment (tapspatel) [reply .github/workflows/build-and-test.yml:249]: resolved on main

File: runtime/tools/python/test/test_check.py (hunk 1 above)
Comment (tapspatel) [reply runtime/tools/python/test/test_check.py]: fixed

File: .github/workflows/build-and-test.yml (hunk 3 above)
Comment (tapspatel) [reply .github/workflows/build-and-test.yml:293]: done

File: runtime/tools/python/ttrt/common/util.py (hunk 2 above)
Comment (tapspatel) [reply runtime/tools/python/ttrt/common/util.py:708]: it's in the pipeline for this commit. You can also view n150_run_results.json etc

**Review by vmilosevic — APPROVED** [review]

**Pull request merged**